File: sample-springboot-login/src/main/java/hello/domain/repository/UserRepository.java
package hello.domain.repository;

import hello.domain.model.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

/**
 * @author archmagece
 * @date 2017-05-12
 */
@Repository
public interface UserRepository extends JpaRepository<UserEntity, Long> {
	UserEntity findByUsername(String username);
}

File: 2022-declarative/domain/jpa/src/main/kotlin/sample/domain/jpa/model/user/UserRepository.kt
package sample.domain.jpa.model.user

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository : JpaRepository<UserEntity, Long> {
    fun findByUsername(username: String): Optional<UserEntity>
    fun findByEmail(email: String): Optional<UserEntity>
}
